moonstone/filtering/mean_filtering.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import plotly.graph_objects as go
+import plotly.io
+from plotly.subplots import make_subplots
+
+from moonstone.analysis.stats import (
+    FilteringStats
+)
+from moonstone.filtering.base import BaseFiltering
+
+logger = logging.getLogger(__name__)
+
+
+class MeanFiltering(BaseFiltering):
+    """
+    This class allows to filter the data according to their mean read count,
+    and remove the data that have a poor mean read count.
+    You can either give a mean read count threshold or the percentage of data
+    that you wish to keep (the threshold will then be computed for you).
+    """
+    def __init__(self, dataframe, threshold=None, percentage_to_keep=0.9):
+        super().__init__(dataframe)
+        self.threshold = threshold
+        if not threshold:
+            self.percentage_to_keep = percentage_to_keep
+
+    def compute_threshold_best_n_percent(self):
+        FS_instance = FilteringStats(self.df)
+        self._items_dict, self._reads_dict = FilteringStats.by_mean(FS_instance)
+        reads_dict_sort = sorted(self._reads_dict.items(), key=lambda x: x[0])  # Sorting by threshold equate to reverse
+        #                                                                        sorting by value but a little quicker
+        for i in reads_dict_sort:
+            if i[1] / self.raw_reads_number > self.percentage_to_keep:  # i[0]:threshold; i[1]:remaining number of reads
+                threshold = float('%.1f' % i[0])
+                reads = int(i[1])
+        logger.info('Computing filtering mean read count threshold...')
+        logger.info('Filtering mean read count threshold set to %.2f.' % threshold)
+        logger.info('Retaining %.1f%% of the data results in %i retained reads.'
+                    % (self.percentage_to_keep * 100, reads))
+        self.threshold = threshold
+        return threshold
+
+    def filter(self):
+        self.steps.append('filtering_by_mean')
+        if not self.threshold:
```

Comment:
Be careful here you should probably use : 

```python
if self.threshold is None:
```

It is then clearer that you are checking the value is None. Also, if for some reason, you would like to apply a threshold of `0` (I don't know why somebody would want to do that...), you would not be able to do so and a new threshold would be automatically computed again.